src/main/java/org/eclipse/basyx/submodel/metamodel/map/submodelelement/operation/OperationVariable.java

```diff
@@ -114,7 +119,33 @@ public void setValue(ISubmodelElement value) {
 			logger.warn("Modeling kind of Operation variable was wrong and automatically changed to ModelingKind.TEMPLATE");
 			HasKind.createAsFacade((Map<String, Object>) value).setKind(ModelingKind.TEMPLATE);
 		}
-		put(Property.VALUE, value);
+
+		Object valueToSet = preprocessValue(value);
+
+		put(Property.VALUE, valueToSet);
+
+	}
+
+	private Object preprocessValue(ISubmodelElement value) {
+		if (value instanceof SubmodelElementCollection) {
+			return handleSMC(value);
+		} else if (value instanceof Entity) {
+			return handleEntity((Entity) value);
+		} else {
+			return value;
+		}
+	}
+
+	private Object handleSMC(ISubmodelElement value) {
+		return SubmodelElementMapCollectionConverter.smElementToMap((SubmodelElementCollection) value);
+	}
+
+	private Object handleEntity(Entity value) {
+		Collection<ISubmodelElement> statements = value.getStatements();
```

Comment:
@FrankSchnicke As far as I correctly understand the SubmodelElementMapCollectionConverter, Entities are processed there, too. Why do you split here?

Reply:
Thanks for pointing this out - I updated the PR.